pkg/performanceprofile/controller/performanceprofile/components/machineconfig/machineconfig.go

```diff
@@ -145,8 +146,14 @@ func getIgnitionConfig(profile *performancev2.PerformanceProfile) (*igntypes.Con
 	}
 
 	// add script files under the node /usr/local/bin directory
+	if profile.Spec.WorkloadHints != nil && profile.Spec.WorkloadHints.RealTime != nil && !*profile.Spec.WorkloadHints.RealTime {
```

Comment:
Maybe is more readable compare with *profile.Spec.WorkloadHints.RealTime also because realTime Hint is default to true

Reply:
we first check if its nil then check the value